pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -76,14 +78,31 @@ type Manager struct {
 }
 
 // NewManager returns a new node shutdown manager.
-func NewManager(getPodsFunc eviction.ActivePodsFunc, killPodFunc eviction.KillPodFunc, shutdownGracePeriodRequested, shutdownGracePeriodCriticalPods time.Duration) *Manager {
-	return &Manager{
+func NewManager(getPodsFunc eviction.ActivePodsFunc, killPodFunc eviction.KillPodFunc, syncNodeStatus func(), shutdownGracePeriodRequested, shutdownGracePeriodCriticalPods time.Duration) (*Manager, lifecycle.PodAdmitHandler) {
+	manager := &Manager{
 		getPods:                         getPodsFunc,
 		killPod:                         killPodFunc,
+		syncNodeStatus:                  syncNodeStatus,
 		shutdownGracePeriodRequested:    shutdownGracePeriodRequested,
 		shutdownGracePeriodCriticalPods: shutdownGracePeriodCriticalPods,
 		clock:                           clock.RealClock{},
 	}
+	return manager, manager
+}
+
+// Admit rejects all pods if node is shutting
+func (m *Manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	m.nodeShuttingDownMutex.Lock()
+	defer m.nodeShuttingDownMutex.Unlock()
+
+	if !m.nodeShuttingDownNow {
+		return lifecycle.PodAdmitResult{Admit: true}
+	}
+	return lifecycle.PodAdmitResult{
+		Admit:   false,
+		Reason:  nodeShutdownReason,
+		Message: nodeShutdownMessage,
```

Comment:
`nodeShutdownMessage` is defined as `Node is shutting, evicting pods` which doesn't seem appropriate as the admission message. 

How about defining a new message, maybe something like `Node is in progress of shutting down; not admitting any new pods`